[Provider-Client] Test test_resource_deletion_during_pvc_pod_creation_and_io.py failing

Test cases in test_resource_deletion_during_pvc_pod_creation_and_io.py are failing with the error given below. 
```
 tests.functional.pv.pv_services.test_resource_deletion_during_pvc_pod_creation_and_io.TestResourceDeletionDuringCreationOperations.test_resource_deletion_during_pvc_pod_creation_and_io[CephBlockPool-mgr]
 Error Details
ocs_ci.ocs.exceptions.TimeoutExpiredError: Timed out after 300s running get("", True, "app=rook-ceph-mgr")
```

```
        # Delete the resource
>       disruption.delete_resource()

tests/functional/pv/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py:326: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
ocs_ci/helpers/disruption_helpers.py:152: in delete_resource
    assert pod_ocp.wait_for_resource(
ocs_ci/ocs/ocp.py:795: in wait_for_resource
    raise (ex)
ocs_ci/ocs/ocp.py:686: in wait_for_resource
    for sample in TimeoutSampler(
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <ocs_ci.utility.utils.TimeoutSampler object at 0x7fd8538c3220>

    def __iter__(self):
        if self.start_time is None:
            self.start_time = time.time()
        while True:
            self.last_sample_time = time.time()
            if self.timeout <= (self.last_sample_time - self.start_time):
                raise self.timeout_exc_cls(*self.timeout_exc_args)
            try:
                yield self.func(*self.func_args, **self.func_kwargs)
            except Exception as ex:
                msg = f"Exception raised during iteration: {ex}"
                log.exception(msg)
            if self.timeout <= (time.time() - self.start_time):
>               raise self.timeout_exc_cls(*self.timeout_exc_args)
E               ocs_ci.ocs.exceptions.TimeoutExpiredError: Timed out after 300s running get("", True, "app=rook-ceph-mgr")

ocs_ci/utility/utils.py:1295: TimeoutExpiredError
```

Multicluster configuration was used with a client as primary cluster.

Example of failed test case: tests/functional/pv/pv_services/test_resource_deletion_during_pvc_pod_creation_and_io.py::TestResourceDeletionDuringCreationOperations::test_resource_deletion_during_pvc_pod_creation_and_io[CephBlockPool-mgr]

Test run - http://magna002.ceph.redhat.com/ocsci-jenkins/openshift-clusters/hcp414-aaa/hcp414-aaa_20240112T084548/multicluster/logs/test_report_1705564011.html

--- a/ocs_ci/helpers/disruption_helpers.py
+++ b/ocs_ci/helpers/disruption_helpers.py
@@ -139,7 +139,9 @@ class Disruptions:
 
     def delete_resource(self, resource_id=0):
         pod_ocp = ocp.OCP(
-            kind=constants.POD, namespace=config.ENV_DATA["cluster_namespace"]
+            kind=constants.POD,
+            namespace=self.resource_obj[resource_id].namespace
+            or config.ENV_DATA["cluster_namespace"],
         )
         if self.cluster_kubeconfig:
             # Setting 'cluster_kubeconfig' attribute to use as the value of the

